hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -428,6 +428,48 @@ def hpat_pandas_series_append_impl(self, to_append):
     return hpat_pandas_series_append_impl
 
 
+@overload_method(SeriesType, 'copy')
+def hpat_pandas_series_copy(self, deep=True):
+    """
+    Pandas Series method :meth:`pandas.Series.groupby` implementation.
+
+    .. only:: developer
+
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_attr2
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_copy_str1
+
+    Parameters
+    -----------
+    self: :class:`pandas.Series`
+        input arg
+    deep: :obj:`bool`, default :obj:`True`
+        Make a deep copy, including a copy of the data and the indices.
+        With deep=False neither the indices nor the data are copied.
+        [HPAT limitations]:
+            - deep=False: shallow copy of index is not supported
+
+    Returns
+    -------
+    :obj:`pandas.Series` or :obj:`pandas.DataFrame`
+        Object type matches caller.
+    """
+    _func_name = 'Method Series.copy().'
+
+    if (isinstance(self, SeriesType) and
+        (isinstance(deep, (types.Omitted, types.Boolean)) or deep == True)
+    ):
+        if isinstance(self.index, types.NoneType):
+            def hpat_pandas_series_copy_impl(self, deep=True):
+                return pandas.Series(self._data.copy() if deep else self._data)
```

Comment:
```suggestion
                if deep:
                    return pandas.Series(self._data.copy())
                else:
                    return pandas.Series(self._data)
```
I think it is more readable.